### PR TITLE
InstallConfig: use secret references instead of strings

### PR DESCRIPTION
**Pull request by csrwng**
Switch to secrets for ssh public key, admin password, and image pull secret

### CODE REVIEW

**Comment by csrwng** [comment]
@dgoodwin ptal

File: config/templates/cluster-deployment.yaml [hunk 1]
```diff
@@ -45,6 +45,30 @@ objects:
     awsAccessKeyId: ${AWS_ACCESS_KEY_ID}
     awsSecretAccessKey: ${AWS_SECRET_ACCESS_KEY}
 
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: ${CLUSTER_NAME}-pull-secret
+  type: kubernetes.io/dockercfg
+  stringData:
+    ".dockercfg": "${PULL_SECRET}"
```
Comment (dgoodwin) [inline config/templates/cluster-deployment.yaml]: The . here makes me a little nervous for bugs, but could we rename to something more like the others, and non-docker specific. (even pull-secret)

File: config/templates/cluster-deployment.yaml [hunk 2]
```diff
@@ -45,6 +45,30 @@ objects:
...
+  stringData:
+    ssh-publickey: "${SSH_KEY}"
+
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: ${CLUSTER_NAME}-admin-creds
+  type: Opaque
+  stringData:
+    password: letmein
```
Comment (dgoodwin) [inline config/templates/cluster-deployment.yaml]: Should be ADMIN_PASSWORD I think.

File: config/templates/cluster-deployment.yaml (hunk 1 above)
Comment (csrwng) [reply config/templates/cluster-deployment.yaml]: Afaik, this is the standard key name for a secret of type "kubernetes.io/dockercfg" (take a look at service account secrets in your cluster). If .dockercfg is not used, I would most likely get a validation error.

File: config/templates/cluster-deployment.yaml (hunk 2 above)
Comment (csrwng) [reply config/templates/cluster-deployment.yaml]: yes :)

**Comment by csrwng** [comment]
Pull updated

**Pull request merged**